tests/test_api_validate.py

```diff
@@ -3101,6 +3118,170 @@ def test_00_registrationtoken_with_validity_period(self):
             self.assertEqual("matching 1 tokens, Outside validity period", detail.get("message"))
 
 
+class RegistrationAndPasswordToken(MyApiTestCase):
+
+    def setUp(self):
+        self.setUp_user_realms()
+
+    def test_00_registration_tokens(self):
+        # Registration tokens always do a genkey, even if we do not set it
+        with self.app.test_request_context('/token/init',
+                                           method='POST',
+                                           data={'user': 'cornelius',
+                                                 'type': 'registration'},
+                                           headers={'Authorization': self.at}):
+            res = self.app.full_dispatch_request()
+            self.assertTrue(res.status_code == 200, res)
+            data = res.json
+            detail = data.get("detail")
+            regcode = detail.get("registrationcode")
+            self.assertEqual(DEFAULT_LENGTH_REG, len(regcode))
+            # Check if a number is contained
+            self.assertRegex(regcode, "[0-9]+")
```

Comment:
Please use `assertRegexpMatches()` here, even though it is deprecated, we can check with Python 2 as well.